apps/balancer-tools/src/app/poolsimulator/(components)/PoolParamsForm.tsx

```diff
@@ -0,0 +1,178 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+import { FieldValues, useForm } from "react-hook-form";
+
+import Button from "#/components/Button";
+import { Input } from "#/components/Input";
+import { Form, FormField } from "#/components/ui/form";
+import {
+  AnalysisData,
+  PoolTypeEnum,
+  usePoolSimulator,
+} from "#/contexts/PoolSimulatorContext";
+import {
+  ECLPSimulatorDataSchema,
+  StableSwapSimulatorDataSchema,
+} from "#/lib/schema";
+
+const schemaMapper = {
+  MetaStable: StableSwapSimulatorDataSchema,
+  GyroE: ECLPSimulatorDataSchema,
+};
+
+interface IInput {
+  name: "swapFee" | "ampFactor" | "alpha" | "beta" | "lambda" | "c" | "s";
```

Comment:
Isn't there a way to get this from all `poolParams` type?

Reply:
Yes! I did this on https://github.com/bleu-studio/balancer-tools/blob/6c00db3d82cdfc1a41bb49c9df9c789283a1847c/apps/balancer-tools/src/app/poolsimulator/(components)/PoolParamsForm.tsx#L23